modules/fedimint-ln-client/src/lib.rs

```diff
@@ -1704,3 +1778,22 @@ fn tweak_user_key<Ctx: Verification + Signing>(
         .add_exp_tweak(secp, &Scalar::from_be_bytes(tweak).expect("can't fail"))
         .expect("tweak is always 32 bytes, other failure modes are negligible")
 }
+
+/// Tweak a secret key with an index, this is used to claim an unspent incoming
+/// contract.
+fn tweak_user_secret_key<Ctx: Verification + Signing>(
+    secp: &Secp256k1<Ctx>,
+    key_pair: KeyPair,
+    index: u64,
+) -> KeyPair {
+    let public_key = key_pair.public_key();
+    let mut hasher = HmacEngine::<sha256::Hash>::new(&public_key.serialize()[..]);
+    hasher.input(&index.to_be_bytes());
+    let tweak = Hmac::from_engine(hasher).into_inner();
+
+    let secret_key = key_pair.secret_key();
+    let sk_tweaked = secret_key
+        .add_tweak(&Scalar::from_be_bytes(tweak).expect("Cant fail"))
+        .expect("Cant fail");
+    KeyPair::from_secret_key(secp, &sk_tweaked)
+}
```

Comment:
Might be good to expose the pub key equivalent so that LNURL server builders have an easier time